lib/manager/util/index.js

```diff
@@ -234,6 +240,92 @@ export function isEditingDisabled (
   )
 }
 
+/**
+ * Checks if the given value is a valid value for a latitude
+ */
+function isValidLatitude (n: number): {
+  error?: string,
+  valid: boolean
+} {
+  if (n <= 90 && n >= -90) {
+    return { valid: true }
+  } else {
+    return {
+      error: `Expected a value between 90 and -90, but found ${n}`,
+      valid: false
+    }
+  }
+}
+
+/**
+ * Checks if the given value is a valid value for a latitude
+ */
+function isValidLongitude (n: number): {
+  error?: string,
+  valid: boolean
+} {
+  if (n <= 180 && n >= -180) {
+    return { valid: true }
+  } else {
+    return {
+      error: `Expected a value between 180 and -180, but found ${n}`,
+      valid: false
+    }
+  }
+}
+
+/**
+ * Attempts to parses a string into a bounds object. The success or failure and
+ * any errors that were encountered are returned.
+ */
+export function parseBounds (val: string): {
+  bounds?: Bounds,
+  errors?: Array<string>,
+  valid: boolean
+} {
+  const errors = []
+
+  function validateVal (val, label, validationFn) {
+    const validationResult = validationFn(val)
+    if (!validationResult.valid) {
+      errors.push(`${label} value is invalid. ${validationResult.error || ''}`)
+    }
+  }
+
+  const bBox = val.split(',')
+    .map(parseFloat)
+    // Filter out any bad parsed values
+    .filter(parsedValue => !isNaN(parsedValue))
+
+  if (bBox.length === 4) {
+    const [west, south, east, north] = bBox
+
+    // make sure all values are valid
+    validateVal(west, 'West', isValidLongitude)
+    validateVal(east, 'East', isValidLongitude)
+    validateVal(north, 'North', isValidLatitude)
+    validateVal(south, 'South', isValidLatitude)
+
+    if (errors.length > 0) {
+      return {
+        errors,
+        valid: false
+      }
+    }
+
+    // valid!
+    const bounds = {west, south, east, north}
+    return {
+      bounds,
+      valid: true
+    }
+  }
+  return {
+    errors: [`Incorrect amount of values. Found ${bBox.length}, expected 4.`],
+    valid: false
+  }
```

Comment:
This might be nitpicky, but I kind of prefer this invalid return statement to come before the above conditional block. Instead of using the `bBox.length === 4` condition to handle the valid number of args, changing it so that the incorrect number of args short circuits the validation check.

```
if (bBox.length !== 4) {
  return {
    errors: [`Incorrect amount of values. Found ${bBox.length}, expected 4.`],
    valid: false
  }
}

const [west, south, east, north] = bBox
// make sure all values are valid
validateVal(west, 'West', isValidLongitude)
validateVal(east, 'East', isValidLongitude)
validateVal(north, 'North', isValidLatitude)
validateVal(south, 'South', isValidLatitude)

if (errors.length > 0) {
  return {
    errors,
    valid: false
  }
}

// valid!
const bounds = {west, south, east, north}
return {
  bounds,
  valid: true
}
```

I can't really explain why that seems preferable to me...

Reply:
It is actually a lot less indentation. Probably saves maybe 10 bytes in the file size.

Reply:
#TrimTheFat